### PR TITLE
Less warnings

### PR DESCRIPTION
**Pull request by ptbrowne**
- Removed expected warnings
- Ignore non problematic warnings

See [issue](https://github.com/cozy/cozy-ui/issues/1318) for more information

### CODE REVIEW

File: react/Field/index.jsx [hunk 1]
```diff
@@ -246,7 +246,8 @@ const allowedTypes = [
   'textarea',
   'text',
   'url',
-  'number'
+  'number',
+  'contact'
```
Comment (y-lohse) [inline react/Field/index.jsx]: Is that a real input type? I've never heard of it and a quick search doesn't return anything.

File: test/jestsetup.js [hunk 2]
```diff
@@ -5,3 +5,56 @@ configure({ adapter: new Adapter() })
...
+
+const ignoreOnConditions = (originalWarn, ignoreConditions) => {
+  return function(...args) {
+    const msg = args[0]
+    if (ignoreConditions.some(condition => condition(msg))) {
+      return
+    }
+    throw new Error(
+      'Code in tests should not emit any warning. Warning emitted:' + msg
+    )
```
Comment (y-lohse) [inline test/jestsetup.js]: Hardcore… but ok I think, it's not too difficult to add exceptions and promotes good practice.

Comment (ptbrowne) [reply test/jestsetup.js]: In fact it was not meant to be in this commit. I prefer to remove it for now and add it later.

File: react/Field/index.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/Field/index.jsx]: Yes it is real :) https://github.com/cozy/cozy-ui/blob/master/react/Field/index.jsx#L182

File: react/__snapshots__/examples.spec.jsx.snap [hunk 3]
```diff
@@ -5754,7 +5767,7 @@ exports[`Spinner should render examples: Spinner 5`] = `
     <div class=\\"styles__c-spinner___1snK7\\"><svg class=\\"styles__icon___23x3R styles__icon--spin___ybfC1\\" style=\\"fill: var(--primaryColor);\\" width=\\"80\\" height=\\"80\\">
         <use xlink:href=\\"#spinner\\"></use>
       </svg>
-      <p>loading.hi</p>
+      <p>loading.helloWorld</p>
```
Comment (Crash--) [inline react/__snapshots__/examples.spec.jsx.snap]: We should have `Hello World !` with your modification, no? (https://github.com/cozy/cozy-ui/pull/1319/files#diff-aae4f4155d6342669aceedd221607d8eR58)

Comment (ptbrowne) [reply react/__snapshots__/examples.spec.jsx.snap]: Yes, this is weird, the Argos screenshot had the change. Looking into it.

Comment (ptbrowne) [reply react/__snapshots__/examples.spec.jsx.snap]: https://github.com/cozy/cozy-ui/pull/1325, thanks good catch

**Pull request merged**